controller/src/beerocks/bml/bml.cpp

```diff
@@ -709,3 +709,60 @@ int bml_start_dcs_single_scan(BML_CTX ctx, const char *radio_mac, int dwell_time
     return pBML->start_dcs_single_scan(tlvf::mac_from_string(std::string(radio_mac)), dwell_time,
                                        channel_pool, channel_pool_size);
 }
+
+int bml_client_get_client_list(BML_CTX ctx, char *client_list, unsigned int *client_list_size)
+{
+    // Validate input parameters
+    if (!ctx && !client_list && !client_list_size) {
+        return (-BML_RET_INVALID_ARGS);
+    }
+
+    auto pBML = static_cast<bml_internal *>(ctx);
+    std::string temp_client_list;
+    int ret = pBML->client_get_client_list(temp_client_list, client_list_size);
+    if (!ret) {
+        beerocks::string_utils::copy_string(client_list, temp_client_list.c_str(),
+                                            *client_list_size);
+        return BML_RET_OK;
+    }
+
+    // Clear the outgoing arguments
+    *client_list_size = 0;
+    *client_list      = '\0';
```

Comment:
I commented on it last time.
You wrote:
>This is a failed scenario, the "returning" values are incorrect
But changed the 0 to '\0' for the client_list.

I didn't notice that inside the `if` block is a success scenario.
Usually, we handle errors, and then continue with the code. So in this case where 0 means success it should look like this:
```cpp
if (ret < 0) {
   // handle error
}
// rest of the code.
```

Also, in case of an error, you should not change the output arguments. The user should not use them either way.

Reply:
Changed to
```
    int ret = pBML->client_get_client_list(temp_client_list, client_list_size);
    if (!ret) {
        beerocks::string_utils::copy_string(client_list, temp_client_list.c_str(),
                                            *client_list_size);
    }

    return ret;
```